mkdocs_simple_plugin/plugin.py

```diff
@@ -405,8 +434,11 @@ def on_pre_build(self, config, **kwargs):
         self.merge_docs_dir = self.config['merge_docs_dir']
         self.semiliterate = []
         for item in self.config['semiliterate']:
-            item['pattern'] = re.compile(item['pattern'])
-            self.semiliterate.append(item)
+            self.semiliterate.append(
+                # Note it is critical that the original item not be modified,
+                # so if this code is changed, the entry added to semiliterate
+                # needs to remain at least a shallow copy of `item`.
+                dict(item, pattern=re.compile(item['pattern'])))
```

Comment:
I don't understand this comment.  Could you explain how this is different from the original?

Reply:
The prior code in this spot modified `item` in place and then added it to semiliterate.This had two drawbacks: (1) since you are displaying the config object for simple in your documentation, it garbled that documentation; (2) if the config object for
simple happens to internally share structure between items, which can happen, then initialization fails. Hence the comment that the value of self.config['semiliterate'] not be modified (although I couched it about the items, since it was the individual
items that were being modified).  If you'd like I could rephrase the comment as:
" Note it is critical that the value of self.config not be modified. Hence, if this code is changed, the entry added to self.semiliterate needs to remain at least a shallow copy of `item`."
(or feel free to change the comment as you see fit). Hopefully I've clarified my concerns here.